core/src/main/java/org/mskcc/cbio/portal/servlet/CancerStudyView.java

```diff
@@ -181,28 +173,42 @@ private boolean buildResponse(HttpServletRequest request)
         }
 
         Set<String> unKnownStudies = new HashSet<>();
-        Set<String> unauthorizedStudies = new HashSet<>();
         
-        //filter unknown and unauthorized studies
-        Set<VirtualStudy> filteredStudies = studies.stream().filter(map -> {
-			for (VirtualStudySamples _study : map.getData().getStudies()) {
-				try {
-					CancerStudy _cancerStudy = getCancerStudyDetails(_study.getId());
-					if (_cancerStudy == null) 
-						unKnownStudies.add(_study.getId());
-					else 
-						return true;
-				} catch (NotAuthorizedException notAuthorizedException) {
-					unauthorizedStudies.add(_study.getId());
-				} catch (DaoException e) {
-					throw new RuntimeException(e);
-				}
-			}
-			return false;
-        }).collect(Collectors.toSet());
+        if(studies.isEmpty()) {
+        		unKnownStudies = inputStudyMap.keySet();
+        	
+        } else {
+            unKnownStudies =  studies
+	        		.stream()
+	        		.filter(obj -> {
+	        			if(inputStudyMap.containsKey(obj.getId())) {
```

Comment:
Isn't studies a subset of inputStudyMap?

Reply:
Yeah true, it should be the opposite here.